src/java/com/puppetlabs/ssl_utils/SSLUtils.java

```diff
@@ -576,6 +576,24 @@ else if (obj instanceof PEMKeyPair)
     }
 
     /**
+     * Decodes the provided object (read from a PEM stream via {@link #pemToObjects}) into a key pair.
+     *
+     * @param obj The object to decode into a PrivateKey
+     * @return The KeyPair decoded from the object
+     * @throws PEMException
+     * @see #pemToKeyPair
+     * @see #pemToKeyPairs
+     */
+    public static KeyPair objectToKeyPair(Object obj)
+        throws PEMException
+    {
+        if (obj instanceof PEMKeyPair)
+            return new JcaPEMKeyConverter().getKeyPair((PEMKeyPair) obj);
```

Comment:
and fwiw, I don't actually care about the name of the method, just used it as an example of what I it seemed to me was actually happening.